lib/puppet/ssl/host.rb

```diff
@@ -419,6 +429,140 @@ def load_crls(path)
     end
   end
 
+  # Ensures that the CA certificate is available for either generating or
+  # validating the host's cert.
+  # It will first check if the cert is present in memory (used for testing),
+  # then check on disk, and finally try to download it.
+  # @return true if the CA certificate was found, false otherwise
+  def ensure_ca_certificate
+    file_path = certificate_location('ca')
+    if check_for_certificate_in_memory('ca')
+      return true
+    elsif Puppet::FileSystem.exist?(file_path)
+      load_certificate_bundle(Puppet::FileSystem.read(file_path))
+      return true
+    else
+      bundle = download_ca_certificate_bundle
+      if bundle
+        save_certificate_bundle(bundle)
+        return true
+      end
+    end
+    false
```

Comment:
This will only be hit if `if bundle` is false, right? In that case we're returning `false` instead of the `nil` that the expression would return? What if we return false explicitly from the above conditional, making it exhaustive, and then removed the early returns?